### DISCUSSION TITLE
AR is not working on Android devices (couldn't load object)

### DISCUSSION THREAD

**Discussion by chandanwebkul** [google/model-viewer#4331] [Q&A]
### Description
I am using version 3.1.4.
It's working at my developer instance but at production instance it's not working.
Showing couldn't load object at android device 
it's working on IOS issue with android device 
with chrome broswer.
I have checked cloudflare, i am not using cloudflare. 

#### Live Demo

https://glitch.com/edit/#!/model-viewer

### Version

- model-viewer: v3.1.1

### Browser Affected

- [x] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [x] WebXR
- [x] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
Can you complete the issue template so that we can help? A repro link (to your site) and your device and browser versions are required info. Also, what AR mode is failing, `webxr` or `scene-viewer`? You should restrict your `ar-modes` to verify.

**Comment by chandanwebkul** [comment]
Thanks elalish to consider my issue 
Android V9+
Browser chrome 
issue : Couldn't load object
Same code execute for the 
IOS V16.x
Borwser: Chrome
Issue : NO issue

I am using the following template
<model-viewer style="margin: 0px auto; display: block;" data-js-focus-visible="" src="https://yodacart.com/shop/pub/media/catalog/product/glbmodels/TVLG006.glb" alt="A 3D model of an product" auto-rotate="" ar="" camera-controls="" touch-action="pan-y" shadow-intensity="1" camera-orbit="33deg 67deg auto" ar_preferred="" ar-modes="scene-viewer webxr quick-look" ar-status="not-presenting">
         <button id="ar-button" slot="ar-button" style="background-color: transparent;border-radius: 25px;
         border: 1px solid darkgray;position: absolute;bottom: 0px;right: 26%;">
            View In Your Space
        </button>
    </model-viewer>

Facing issue with AR when click on the button view in your space 

it's showing object couldn't load
same code is running at two instance one of the working another is not 
and everything is same at both instance.

**Comment by chandanwebkul** [comment]
```html
<model-viewer style="margin: 0px auto; display: block;" data-js-focus-visible="" src="https://yodacart.com/shop/pub/media/catalog/product/glbmodels/TVLG006.glb" alt="A 3D model of an product" auto-rotate="" ar="" camera-controls="" touch-action="pan-y" shadow-intensity="1" camera-orbit="33deg 67deg auto" ar_preferred="" ar-modes="scene-viewer webxr quick-look" ar-status="not-presenting">
         <button id="ar-button" slot="ar-button" style="background-color: transparent;border-radius: 25px;
         border: 1px solid darkgray;position: absolute;bottom: 0px;right: 26%;">
            View In Your Space
        </button>
    </model-viewer>
```

**Reply by chandanwebkul** [reply]
```javascript
 setTimeout(() => {
            var link = document.getElementById('ar-button');
            link.click();
            console.log("click evnet occur");
        }, 3000);
```

**Comment by chandanwebkul** [comment]
I also use click event on the custom ar-button for lunch the Ar in mobile device
please check the i hope it help you
```html

``` 
```javascript
requirejs(['jquery',
'Webkul_WebAr/js/webxr/mv/model-viewer.min',
],function($,umd) {
        var productType = '<?= /* @noEscape */ $productType?>';
        const modelViewerVariants = document.querySelector("model-viewer");
        modelViewerVariants.addEventListener("load", (ev) => {
            if(productType == 'configurable') {
                if ($('.swatch-opt').length > 0) {
                $('body').on('click', '.swatch-option', function(e) {
                var swatchid = $('.swatch-attribute').attr('data-option-selected');
                    if (swatchid) {
                        var label = $(".swatch-option[data-option-id='" +swatchid+"']").attr('data-option-label');
                       modelViewerVariants.variantName = label;
                    }  
              });
            } else {
                const select = document.querySelector('.product-options-wrapper select[id^="attribute"]');
                select.addEventListener('input', (event) => {
                        var optionText="";var optionTextArr =[]; 
                        var selectedText = $("select option:selected").text();
                        if(selectedText.indexOf('+') == -1) {
                            optionText = selectedText;
                        } else {
                            optionTextArr =  selectedText.split('+');
                            optionText = $.trim(optionTextArr[0]);
                        }
                    
                        modelViewerVariants.variantName = optionText;
                });
            }
            }
});
    var status = '<?= /* @noEscape */$status?>';
    if(status){
        setTimeout(() => {
            var link = document.getElementById('ar-button');
            link.click();
            console.log("click evnet occur");
        }, 3000);
    }
});
```

**Reply by elalish** [reply]
Try `ar-modes="webxr quick-look"` - if the AR button doesn't show up, please check the JS console and list anything it says here.
